kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableWritesSuite.scala

```diff
@@ -1009,4 +1004,228 @@ class DeltaTableWritesSuite extends DeltaTableWriteSuiteBase with ParquetSuiteBa
     schema.foreach(s => txnBuilder = txnBuilder.withSchema(engine, s))
     txnBuilder.build(engine)
   }
+
+  test("create table with unsupported column mapping mode") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val ex = intercept[InvalidConfigurationValueException] {
+        createTxn(engine, tablePath, isNewTable = true, testSchema, partCols = Seq.empty,
+          tableProperties = Map(TableConfig.COLUMN_MAPPING_MODE.getKey -> "invalid"))
+          .commit(engine, emptyIterable())
+      }
+      assert(ex.getMessage.contains("Invalid value for table property " +
+        "'delta.columnMapping.mode': 'invalid'. Needs to be one of none, id, name."))
+    }
+  }
+
+  test("create table with column mapping mode = none") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      createTxn(engine, tablePath, isNewTable = true, testSchema, partCols = Seq.empty,
+        tableProperties = Map(TableConfig.COLUMN_MAPPING_MODE.getKey -> "none"))
+        .commit(engine, emptyIterable())
+
+      val table = Table.forPath(engine, tablePath)
+      assert(table.getLatestSnapshot(engine).getSchema(engine).equals(testSchema))
+    }
+  }
+
+  test("update table with unsupported column mapping mode") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = Table.forPath(engine, tablePath)
+      createTxn(engine, tablePath, isNewTable = true, testSchema, Seq.empty)
+        .commit(engine, emptyIterable())
+
+      val ex = intercept[InvalidConfigurationValueException] {
+        table.createTransactionBuilder(engine, testEngineInfo, Operation.WRITE)
+          .withTableProperties(
+            engine,
+            Map(ColumnMapping.COLUMN_MAPPING_MODE_KEY -> "invalid").asJava)
+          .build(engine)
+      }
+      assert(ex.getMessage.contains("Invalid value for table property " +
+        "'delta.columnMapping.mode': 'invalid'. Needs to be one of none, id, name."))
+    }
+  }
+
+  test("update table with unsupported column mapping mode change") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = Table.forPath(engine, tablePath)
+      createTxn(engine, tablePath, isNewTable = true, testSchema, partCols = Seq.empty,
+        tableProperties = Map(TableConfig.COLUMN_MAPPING_MODE.getKey -> "name"))
+        .commit(engine, emptyIterable())
+
+      val ex = intercept[IllegalArgumentException] {
+        table.createTransactionBuilder(engine, testEngineInfo, Operation.WRITE)
+          .withTableProperties(
+            engine,
+            Map(TableConfig.COLUMN_MAPPING_MODE.getKey -> "none").asJava)
+          .build(engine)
+      }
+      assert(ex.getMessage.contains("Changing column mapping mode " +
+        "from 'name' to 'none' is not supported"))
+    }
+  }
+
+  test("update column mapping mode from id to name on existing table") {
```

Comment:
`cannot update column mapping mode from id to name on existing table`

Reply:
makes sense, done